apps/total-typescript/src/pages/invoices/[merchantChargeId].tsx

```diff
@@ -2,24 +2,36 @@ import * as React from 'react'
 import {DownloadIcon} from '@heroicons/react/outline'
 import {convertToSerializeForNextResponse} from '@skillrecordings/commerce-server'
 import {useLocalStorage} from 'react-use'
-import {GetServerSideProps} from 'next'
+import {GetServerSideProps, InferGetServerSidePropsType} from 'next'
 import {stripe} from '@skillrecordings/commerce-server'
-import {Coupon, getSdk, MerchantProduct} from '@skillrecordings/database'
+import {Coupon, getSdk} from '@skillrecordings/database'
 import {Stripe} from 'stripe'
 import fromUnixTime from 'date-fns/fromUnixTime'
 import Layout from 'components/app/layout'
 import format from 'date-fns/format'
-import {prisma} from '@skillrecordings/database'
+// combine this line with the same-package import above
+import {prisma, Product} from '@skillrecordings/database'
 import {getCurrentAbility} from '@skillrecordings/ability'
 import {getToken} from 'next-auth/jwt'
+import {isArray} from 'lodash'
 
-export const getServerSideProps: GetServerSideProps = async ({
+type InvoiceProps = {
+  charge: Stripe.Charge
+  product: Product
+  merchantChargeId: string
+  bulkCoupon: Coupon | null
+}
```

Comment:
Define the types in one place at the top of the page, then use them in `getServerSideProps` with `GetServerSideProps<InvoiceProps>`. Then the page component can infer the props it is receiving by doing `InferGetServerSidePropsType<typeof getServerSideProps>`.